api/monitor/monitorclient/client.go

```diff
@@ -191,8 +196,26 @@ func (c *componentClient) Send(msg interface{}) {
 	}
 }
 
+var stableTimeout = 10 * time.Second
+
 func (c *componentClient) run(monitor, stream string, buf *streambuffer.Stream) {
-	if err := buf.Run(); err != nil && err != context.Canceled {
-		c.log.WithField("Monitor", monitor).WithError(err).Warnf("%s stream failed", buf)
+	var t *time.Timer
+	var streamErrors int32
+	for {
+		t = time.AfterFunc(stableTimeout, func() { atomic.SwapInt32(&streamErrors, 0) })
+		err := buf.Run()
+		t.Stop()
```

Comment:
https://godoc.org/time#Timer.Stop
this is potential leak I assume, you should explicitly drain the channel